account_reconcile_model_multicompany_propagate/tests/test_reconcile_model.py

```diff
@@ -0,0 +1,162 @@
+                        },
+                    ),
+                    (
+                        0,
+                        0,
+                        {
+                            "partner_id": cls.partner_2.id,
+                            "payment_ref_regex": "payment ref regex 2",
+                            "narration_regex": "narration regex 2",
+                        },
+                    ),
+                ],
+            }
+        )
+        cls.tax_sale_b_company_2 = cls.safe_copy(cls.company_data_2["default_tax_sale"])
+
+    def test_propagate_to_other_companies(self):
+        self.rule_1.propagate_to_other_companies()
+        rule_1_company_2 = self.env["account.reconcile.model"].search(
+            [
+                ("name", "=", self.rule_1.name),
+                ("company_id", "=", self.company_data_2["company"].id),
+            ]
+        )
+        self.assertTrue(rule_1_company_2)
+        ReconcileModel = self.env["account.reconcile.model"]
+        # Check propagate fields
+        for field in ReconcileModel._propagated_fields():
+            self.assertEqual(
+                getattr(rule_1_company_2, field), getattr(self.rule_1, field)
+            )
+        # Check line_ids
+        self.assertEqual(
+            rule_1_company_2.line_ids.mapped("sequence"),
+            self.rule_1.line_ids.mapped("sequence"),
+        )
+        self.assertEqual(
+            rule_1_company_2.line_ids.account_id.mapped("code"),
+            self.rule_1.line_ids.account_id.mapped("code"),
+        )
+        self.assertEqual(
+            rule_1_company_2.line_ids.mapped("label"),
+            self.rule_1.line_ids.mapped("label"),
+        )
+        self.assertEqual(
+            rule_1_company_2.line_ids.mapped("amount_type"),
+            self.rule_1.line_ids.mapped("amount_type"),
+        )
+        self.assertEqual(
+            rule_1_company_2.line_ids.mapped("force_tax_included"),
+            self.rule_1.line_ids.mapped("force_tax_included"),
+        )
+        self.assertEqual(
+            rule_1_company_2.line_ids.mapped("amount_string"),
+            self.rule_1.line_ids.mapped("amount_string"),
+        )
+        self.assertEqual(
+            rule_1_company_2.line_ids.tax_ids.mapped("name"),
+            self.rule_1.line_ids.tax_ids.mapped("name"),
+        )
+        # Check partner_mapping_line_ids
+        self.assertEqual(
+            rule_1_company_2.partner_mapping_line_ids.mapped("partner_id"),
+            self.rule_1.partner_mapping_line_ids.mapped("partner_id"),
+        )
+        self.assertEqual(
+            rule_1_company_2.partner_mapping_line_ids.mapped("payment_ref_regex"),
+            self.rule_1.partner_mapping_line_ids.mapped("payment_ref_regex"),
+        )
+        self.assertEqual(
+            rule_1_company_2.partner_mapping_line_ids.mapped("narration_regex"),
+            self.rule_1.partner_mapping_line_ids.mapped("narration_regex"),
+        )
```

Comment:
IMHO it would be better to replace all these assertions for just one call, similar to this:

```suggestion
        self.assertRecordValues(
            rule_1_company_2,
            [{
                "sequence": "1000",
                "rule_type": "invoice_matching",
                ...,
                "partner_mapping_line_ids": [
                    {
                        "partner_id": cls.partner_1.id,
                        "payment_ref_regex": "payment ref regex",
                        "narration_regex": "narration regex",
                    },
                    ...,
                ],
            }]
        )
```

The benefits of having it hardcoded is that, instead of implicitly testing that all the calls to `mapped()` do what you expect them to do, you assert the final results. Also, the test becomes more readable.

However, this is just a enhancement suggestion. The test is good. 😉

Reply:
You're right. I changed test.